src/registration/Register.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { ReactNode, useState } from 'react';
+import { refreshSamUserAttributes, refreshTerraProfile, signOut } from 'src/auth/auth';
+import { ButtonPrimary, ButtonSecondary, LabeledCheckbox } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import planet from 'src/images/register-planet.svg';
+import { Ajax } from 'src/libs/ajax';
+import { SamUserAttributes } from 'src/libs/ajax/User';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import Events from 'src/libs/events';
+import { FormLabel } from 'src/libs/forms';
+import { RegistrationLogo } from 'src/libs/logos';
+import { authStore, getTerraUser, TerraUser } from 'src/libs/state';
+import { CommunicationPreferencesCheckbox } from 'src/registration/CommunicationPreferencesCheckbox';
+import { InterestInTerraCheckbox } from 'src/registration/InterestInTerraCheckbox';
+import { LabelledTextInput } from 'src/registration/LabelledTextInput';
+import validate from 'validate.js';
+
+const constraints = ({ partOfOrganization }: { partOfOrganization: boolean }) => {
+  return {
+    givenName: { presence: { allowEmpty: false } },
+    familyName: { presence: { allowEmpty: false } },
+    email: { presence: { allowEmpty: false } },
+    institute: { presence: { allowEmpty: !partOfOrganization } },
+    department: { presence: { allowEmpty: !partOfOrganization } },
+    title: { presence: { allowEmpty: !partOfOrganization } },
+  };
+};
+
+export const Register = (): ReactNode => {
+  const user: TerraUser = getTerraUser();
+  const userAttributes: SamUserAttributes = authStore.get().terraUserAttributes;
```

Comment:
I'm a bit apprehensive about this getting the store's value once instead of subscribing to it with `useStore`, but if you're sure this isn't updated by anything else while Register is displayed, then this may be fine.

Reply:
This is a bad way of getting the default `true` value. Removed.